dspace-api/src/main/java/org/dspace/discovery/SolrServiceImpl.java

```diff
@@ -2305,4 +2597,64 @@ public String calculateExtremeValue(Context context, String valueField,
         }
         return null;
     }
+
+    private void addNamedResourceTypeIndex(SolrInputDocument document, String acvalue, String fvalue) {
+
+        document.addField("namedresourcetype_filter", acvalue);
+
+        String[] avalues = acvalue.split(SolrServiceImpl.AUTHORITY_SEPARATOR);
+        acvalue = avalues[0];
+
+        String avalue = avalues[1];
+        document.addField("namedresourcetype_authority", avalue);
+        document.addField("namedresourcetype_group", avalue);
+        document.addField("namedresourcetype_ac", acvalue);
+
+        Pattern pattern = Pattern.compile(REGEX);
+        Matcher matcher = pattern.matcher(acvalue);
+        if (matcher.matches()) {
+            fvalue = matcher.group(1);
+        }
+
+        document.addField("namedresourcetype_keyword", fvalue);
+    }
+
+    class IndexerThread extends Thread {
```

Comment:
Should really implement the Runnable class instead of extending thread as this is the preferred way to work with threads in java.

https://stackoverflow.com/questions/541487/implements-runnable-vs-extends-thread-in-java

Reply:
I agree with the suggestion but support for multi-threads has been withdrawn from this PR as suggested by @cwilper https://github.com/4Science/DSpace/pull/65#issuecomment-477177919 and will be eventually introduced in a separate PR/ticket